src/lib.rs

```diff
@@ -53,6 +53,38 @@ fn fail_derive(s: synstructure::Structure) -> quote::Tokens {
         }
     });
 
+    let name = &s.ast().ident;
+
+    #[cfg(feature = "std")]
+    let box_fail = quote! {
+        impl ::failure::Fail for Box<#name>
+            where #name: ::failure::Fail
+        {
+            fn cause(&self) -> ::std::option::Option<&::failure::Fail> {
+                (**self).cause()
+            }
+
+            fn backtrace(&self) -> ::std::option::Option<&::failure::Backtrace> {
+                (**self).backtrace()
+            }
+        }
+    };
+
+    #[cfg(not(feature = "std"))]
+    let box_fail = quote! {
+        impl ::failure::Fail for Box<#name>
```

Comment:
If it's no_std, where is Box coming from?

Reply:
@cramertj is right; we should just not provide this impl in `no_std`.